lib/semian/net_http.rb

```diff
@@ -0,0 +1,85 @@
+require 'semian'
+require 'semian/adapter'
+require 'net/https'
+require 'open-uri'
+
+module Net
+  ::Net::ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  Net::ResourceBusyError = Class.new(SemianError)
+  Net::CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    DEFAULT_SEMIAN_OPTIONS = @raw_semian_options = {
```

Comment:
This is confusing, you have a class instance variable here called `@raw_semian_options` and below you have an instance variable which is also called `@raw_semian_options`. Those two are not the same! I'm gonna assume that wasn't on purpose?